app/src/test/java/com/andryoga/safebox/worker/BackupDataWorkerTest.kt

```
@@ -0,0 +1,262 @@
+        assertThat(result).isEqualTo(Result.success())
+        assertThat(fakeBackupMetadataRepo.updatedDate).isNotNull()
+
+        val createdFiles = tempDir.listFiles { f -> f.name.startsWith("SafeBoxBackup") }
+        assertThat(createdFiles).isNotNull()
+        assertThat(createdFiles!!.isNotEmpty()).isTrue()
+    }
+
+    @Test
+    fun doWork_whenExtraBackupsExist_deletesOldestFilesExceedingLimit() = runTest {
+        for (i in 1..(CommonConstants.MAX_BACKUP_FILES + 2)) {
+            val formatted = String.format("%02d", i)
```

detekt (1.23.8)

[warning] String.format("%02d", i) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)